### ISSUE TITLE
reasonable wavernn loss range?

### ISSUE THREAD

**Issue by shuzeZHAO** [fatchord/WaveRNN#100]
Hi all, I have some basic questions on how to train WaveRNN. 

I'm wondering what's the final loss you got when you train WaveRNN, and how many steps does it take for the model to output human-like voices? My loss stuck at around 3.5 and won't go lower. I trained the model with our own dataset, when I run gen_tacotron.py, I got mostly noise with some vague human voice in the background, I'm trying to find out where is the problem. 

I'm not sure if I trained the model correctly. Basically, I trained the tacotron model first, after it's done it generates GTA files. Then I started training WaveRNN using the GTA files, it will take a few days. Just want to confirm if this is the right way to train these two models.  

Thanks!
Shuze

**Comment by shuzeZHAO** [comment]
a follow-up question. I saw the mel_outputs and linear are generated slightly different than the Tacotron2 paper, am I correct? 

`        # Concat the mel outputs into sequence
        mel_outputs = torch.cat(mel_outputs, dim=2)
        
        # Post-Process for Linear Spectrograms
        postnet_out = self.postnet(mel_outputs)
        linear = self.post_proj(postnet_out)
        linear = linear.transpose(1, 2)
        
        # For easy visualisation
        attn_scores = torch.cat(attn_scores, 1)
        attn_scores = attn_scores.cpu().data.numpy()
            
        return mel_outputs, linear, attn_scores`

The gta is from the second output of model, which is the linear output. Is this a mistake? I thought gta should be mel-spectrum. 

train_tacotron.py:     
`        _, gta, _ = model(x, mels)`

Sorry for dropping so many questions.

**Comment by fatchord** [comment]
@shuzeZHAO  It all depends. Whether you use MOL, 8-bits, 9-bits and or mu law will all affect the loss - as well as the actual dataset itself. There are just too many factors involved to give a ballpark number.

If you are new to this model I would recommend training the fastest version which is raw 8-bits without mu-law. This will converge in under 250,000 steps which is achievable in around 24hours on a single 1080gtx. It should start sounding pretty good well before 200,000 steps. There will be some quantisation noise in the samples but at least you know everything is working ok.

Ultimately I would advise you avoid training an MOL model until you are happy with an 8 or 9 bit model. 

The reason for this advice is really just for your own sanity - nothing is more annoying than training a model to over a million steps and finding it doesn't work right in the end.

**Comment by shuzeZHAO** [comment]
> @shuzeZHAO It all depends. Whether you use MOL, 8-bits, 9-bits and or mu law will all affect the loss - as well as the actual dataset itself. There are just too many factors involved to give a ballpark number.
> 
> If you are new to this model I would recommend training the fastest version which is raw 8-bits without mu-law. This will converge in under 250,000 steps which is achievable in around 24hours on a single 1080gtx. It should start sounding pretty good well before 200,000 steps. There will be some quantisation noise in the samples but at least you know everything is working ok.
> 
> Ultimately I would advise you avoid training an MOL model until you are happy with an 8 or 9 bit model.
> 
> The reason for this advice is really just for your own sanity - nothing is more annoying than training a model to over a million steps and finding it doesn't work right in the end.

@fatchord Thank you for your detailed reply. 

Yes, I tried that and the model starts to generate human voice. I will explore your model in details. Thanks again!

**Comment by fatchord** [comment]
@shuzeZHAO You're very welcome. Cheers!
